pkg/asset/installconfig/powervs/client.go

```diff
@@ -0,0 +1,116 @@
+package powervs
+
+import (
+	"os"
+	"strings"
+	 gohttp "net/http"
+
+	"github.com/IBM-Cloud/bluemix-go"
+	"github.com/IBM-Cloud/bluemix-go/api/account/accountv2"
+	"github.com/IBM-Cloud/bluemix-go/authentication"
+	"github.com/IBM-Cloud/bluemix-go/http"
+	//"github.com/IBM-Cloud/bluemix-go/models"
+	"github.com/IBM-Cloud/bluemix-go/rest"
+	bxsession "github.com/IBM-Cloud/bluemix-go/session"
+	//"github.com/dgrijalva/jwt-go"
+	"github.com/form3tech-oss/jwt-go"
+	"github.com/pkg/errors"
+)
+
+type Client struct {
+	*bxsession.Session
+	User               *User
+	AccountAPIV2	   accountv2.Accounts
+}
+
+type User struct {
+	ID         string
+	Email      string
+	Account    string
+}
+
+func authenticateAPIKey(sess *bxsession.Session) error {
+	config := sess.Config
+	tokenRefresher, err := authentication.NewIAMAuthRepository(config, &rest.Client{
+		DefaultHeader: gohttp.Header{
+			"User-Agent": []string{http.UserAgent()},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	return tokenRefresher.AuthenticateAPIKey(config.BluemixAPIKey)
+}
+
+func fetchUserDetails(sess *bxsession.Session) (*User, error) {
+	config := sess.Config
+	user := User{}
+	var bluemixToken string
+
+	if strings.HasPrefix(config.IAMAccessToken, "Bearer") {
+		bluemixToken = config.IAMAccessToken[7:len(config.IAMAccessToken)]
+	} else {
+		bluemixToken = config.IAMAccessToken
+	}
+
+	token, err := jwt.Parse(bluemixToken, func(token *jwt.Token) (interface{}, error) {
+		return "", nil
+	})
+	if err != nil && !strings.Contains(err.Error(), "key is of invalid type") {
+		return &user, err
+	}
+
+	claims := token.Claims.(jwt.MapClaims)
+	if email, ok := claims["email"]; ok {
+		user.Email = email.(string)
+	}
+	user.ID = claims["id"].(string)
+	user.Account = claims["account"].(map[string]interface{})["bss"].(string)
+
+	return &user, nil
+}
+
+func NewClient() (*Client, error) {
+	var apikey string
+	c := &Client{}
+
+	if apikey = os.Getenv("IBMID_PASSWORD"); len(apikey) == 0 {
```

Comment:
can you update this to the API key var? See https://github.com/openshift-powervs/installer/pull/26#discussion_r677851761 for some background

Reply:
Ok